drivers/usb/tests/TESTS/usb_device/basic/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+include(mbed_greentea)
+
+if(NOT "USB_DEVICE_TESTS=1" IN_LIST MBED_TARGET_DEFINITIONS)
```

Comment:
`USB_DEVICE_TESTS` - Looks like another macro to disable tests, to save time in CI perhaps. Doc: https://github.com/ARMmbed/mbed-os/blob/331473a70619afbef10832236bd6085bc5cf5f54/drivers/usb/tests/TESTS/usb_device/README.md#running-tests.

See previous discussion https://github.com/ARMmbed/mbed-os/pull/14986#discussion_r687699374

Reply:
Okay, looks like this has additional hardware requirement - target's own USB (not just DAPLink) connected, which may not have been set up in CI. So not just about saving time which I said earlier.

The user does need a way to pass it. Maybe a CMake option for this?

Reply:
Hi @ARMmbed/mbed-os-test, we have a few Ethernet-connected boards in CI to test the network interface. Any chance we similarly have USB connected boards? Ones with own USB, not just DAPLink? Thanks.

Reply:
I checked with @jamesbeyond. "The USB tests require two USB cables connected to the target from the board farm dispatcher hosts, and this is not supported with RaaS."

So it appears we don't run these tests in CI at the moment.

Reply:
Fair enough. If a user wants to manually enable the USB tests, I guess the only way is to write a json with `"macros": ["USB_DEVICE_TESTS=1"]` and pass it to mbed-tools using `--app-config`.

Reply:
Alternatively, we modify drivers/usb/mbed_lib.json to
```json
{
    "name": "drivers-usb",
    "config": {
        "enable_test": {
            "value": false
        }
    }
}
```
Then they can do in their own json file passed via --app-config:
```
{
    target_overrides: {
        "*": {
            "drivers-usb.enable_test": true
        }
    }
}
```
Maybe the second json can be provided as TESTS/configs/usb.json, which is what we do for bare metal, mbedtls and experimental (PSA).

Reply:
Now I tend not to overcomplicate things, as the existing `EXCLUDE_FROM_ALL` looks fine to me. Marking this conversation as resolved.